plugins/outputs/nats/nats.go

```diff
@@ -85,7 +100,47 @@ func (n *NATS) Connect() error {
 
 	// try and connect
 	n.conn, err = nats.Connect(strings.Join(n.Servers, ","), opts...)
+	if n.Jetstream != nil {
+		// connect to jetstream
+		n.jetstreamClient, err = jetstream.New(n.conn)
+		if err != nil {
+			return fmt.Errorf("failed to connect to jetstream: %w", err)
+		}
+		if n.Jetstream.Stream == "" {
+			return errors.New("stream cannot be empty")
+		}
+		streamExists := n.streamExists(n.Jetstream.Stream)
+		if !streamExists {
+			if !n.Jetstream.AutoCreateStream {
+				return fmt.Errorf("stream %s does not exist", n.Jetstream.Stream)
+			}
+			streamConfigJSON := strings.TrimSpace(n.Jetstream.StreamJSON)
+			var streamCfg jetstream.StreamConfig
+			if len(streamConfigJSON) > 0 {
+				err = json.Unmarshal([]byte(streamConfigJSON), &streamCfg)
+				if err != nil {
+					return fmt.Errorf("invalid jetstream config %w", err)
+				}
+			}
```

Comment:
These options should really be TOML, what about it's own [table](https://toml.io/en/v1.0.0#table)